core/src/main/java/com/linecorp/armeria/common/DefaultHttpHeaders.java

```diff
@@ -186,32 +186,41 @@ public HttpHeaders status(HttpStatus status) {
     @Nullable
     @Override
     public MediaType contentType() {
-        final MediaType contentType = this.contentType;
-        if (contentType != null) {
-            return contentType;
-        }
-
-        final String contentTypeString = get(HttpHeaderNames.CONTENT_TYPE);
-        if (contentTypeString == null) {
-            return null;
-        }
-
-        try {
-            this.contentType = MediaType.parse(contentTypeString);
-            return this.contentType;
-        } catch (IllegalArgumentException unused) {
-            // Invalid media type
-            return null;
-        }
+        return contentType;
     }
 
     @Override
     public HttpHeaders contentType(MediaType contentType) {
         requireNonNull(contentType, "contentType");
-        this.contentType = contentType;
         return set(HttpHeaderNames.CONTENT_TYPE, contentType.toString());
     }
 
+    @Override
+    public HttpHeaders set(AsciiString name, String value) {
+        super.set(name, value);
+        setIfContentType(name, value);
```

Comment:
Every `set()` call will involve string comparison even if the header name is not `content-type`, which is an overkill. We could just do this:

```java
public MediaType contentType() {
    final MediaType contentType = this.contentType;
    final String contentTypeString = get(HttpHeaderNames.CONTENT_TYPE);
    if (contentTypeString == null) {
        return null;
    }
    if (contentType != null && contentType.toString().equals(contentTypeString)) {
        return contentType;
    }
    ...
}
```

Reply:
Fixed Thank you~!